StortSpelProjekt/Tutorials.md

```diff
@@ -195,6 +195,113 @@ exclusive fullscreen may save a couple of your frames per second and is therefor
 exclusive fullscreen mode will also override the window size settings until you loose focus by, for example, pressing the *alt+enter* combination on
 your keyboard. While doing so, the exclusive fullscreen will be changed to a window which will have the size which is decided in the config.txt file.
 
+## Upgrades
+### Making new Upgrades
+To make a new upgrade you need to make a new class that inherits from **Upgrade.h**. 
+In the constructor of this class you need to set the **name** of the class as well as its **type**. 
+The namingconvention we have chosen is to name it the same as the class itself. 
+When it comes to types there are three of them. **PLAYER**, **RANGE** and **ENEMYSPECIFIC**.
+**RANGE** is for when the upgrade has to go on projectiles, **PLAYER** on player/enemy and **ENEMYSPECIFIC** are only for enemies.
+An example of an upgrade constructor:
+
+```cpp
+	UpgradeMeleeTest::UpgradeMeleeTest(Entity* parentEntity) : Upgrade(parentEntity)
+	{
+		SetName("UpgradeMeleeTest");
+		SetType(F_UpgradeType::PLAYER);
+		m_DamageChange = 2;
+	}
+```
+
+An uppgrade has many inherited functions such as OnHit(), ApplyStat() or OnDamage().
+It is using these functions that you decide where/what your upgrade will affect. 
+As an Example take UpgradeRangeTest which will have an emmidiate effect on player health in its ApplyStat() function,
+as well as making projectiles shoot upwards when hitting something in the function OnRangeHit().
+
+```cpp
+	void UpgradeRangeTest::OnRangedHit()
+	{
+		m_pParentEntity->GetComponent<component::AccelerationComponent>()->SetAccelerationDirection(m_Direction);
+		m_pParentEntity->GetComponent<component::AccelerationComponent>()->SetAccelerationSpeed(m_AccelerationSpeed);
+	}
+
+	void UpgradeRangeTest::ApplyStat()
+	{
+		if (m_pParentEntity->HasComponent<component::HealthComponent>())
+		{
+			m_pParentEntity->GetComponent<component::HealthComponent>()->ChangeHealth(m_HealthChange);
+		}
+	}
+```
+
+If an upgrade is bought more than once its level should increase in the function **IncreaseLevel()**.

```

Comment:
When was this decided? It is not in the decision document. Did the game designers decide this in their last meeting?

Reply:
There have been some talks about this, I will try and raise this question for their next meeting to get an official decision. Please join in if you have some opinions or ideas about it!